### DISCUSSION TITLE
Proposal - Allow mixed case attribute names

### DISCUSSION THREAD

**Discussion by jmooring** [yuin/goldmark#279] [Ideas]
Currently, this does not do anything:

````
```text {LineNos=true LineNoStart=42}
foo
```
````

You must do this instead:

````
```text {linenos=true linenostart=42}
foo
```
````

Users are accustomed to case-insensitive HTML attribute names. I understand this is markdown, not HTML, but there is a similar expectation from non-developer users.

Reference:
- <https://github.com/yuin/goldmark-highlighting/issues/30>

**Comment by yuin** [comment]
FYI, pandoc does *NOT* allow this kind of 'relaxed' attribute names.

`````
~~~~ {#mycode .haskell .numberLines startFrom="100"}
qsort []     = []
qsort (x:xs) = qsort (filter (< x) xs) ++ [x] ++
               qsort (filter (>= x) xs)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
`````

works, 

`````
~~~~ {#mycode .haskell .numberLines startfrom="100"}
qsort []     = []
qsort (x:xs) = qsort (filter (< x) xs) ++ [x] ++
               qsort (filter (>= x) xs)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
`````

not works

**Reply by jmooring** [reply]
Sure, but that doesn't mean it's the right thing to do.
